engine/src/main/resources/assets/shaders/lightGeometryPass_frag.glsl

```diff
@@ -143,21 +147,19 @@ void main() {
 
 #if defined (FEATURE_LIGHT_POINT)
 
-
-    //ref: https://imdoingitwrong.wordpress.com/2011/01/31/light-attenuation/
-    // TODO: Make cutoff a shader parameter
-    const float cutoff= 0.005;
-    const float radius = 3.0;
     // calculate basic attenuation
-    float denom = lightDist/radius + 1;
+    // realistic attenuation ref: https://imdoingitwrong.wordpress.com/2011/01/31/light-attenuation/
+    float denom = lightDist/lightAttenuationRange + 1;
     float attenuation = 1.0/(denom*denom);
-    //float attenuation = clamp (1.0 - ((lightDist * lightDist) / (lightAttenuationRange * lightAttenuationRange)), 0.0, 1.0);
-    //float attenuation = 1.0 - (pow (lightDist, lightAttenuationFalloff) / lightAttenuationRange);
 
-    attenuation = (attenuation - cutoff)/(1.0-cutoff);
+    // Force the light to gradually come to a complete stop at the attenuation range + falloff distance.
+    float lightDistPastRange = max(lightDist - lightAttenuationRange, 0.0);
+    float falloffTerm = 1.0 - min(lightDistPastRange / lightAttenuationFalloff, 1.0);
+    attenuation *= falloffTerm;
```

Comment:
This falloff mechanism allows simulating darkness game mechanics by cutting off the light after a certain distance.  One can still have light behave as if in a vacuum by setting the falloff value really high from the LightComponent.

Before:
![terasology-160408120909-1280x720](https://cloud.githubusercontent.com/assets/5922109/14394854/2b093eca-fd83-11e5-889c-7ad76d9ce6e9.jpg)

After with a falloff value of 1.25:
![terasology-160408120925-1280x720](https://cloud.githubusercontent.com/assets/5922109/14394857/37c24396-fd83-11e5-9aa0-d92ee4f49965.jpg)